src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -204,3 +206,115 @@ def validate_infrastructure_type(infra):
         if s.lower() == infra.lower():
             return s
     return "generic"
+
+
+def get_values_file():
+    values_file_provided = False
+    values_file = os.getenv('HELMVALUESPATH')
+    if (values_file is not None) and (os.path.isfile(values_file)):
+        values_file_provided = True
+        logger.warning("Values files detected. Reading additional helm parameters from same.")
+        # trimming required for windows os
+        if (values_file.startswith("'") or values_file.startswith('"')):
+            values_file = values_file[1:]
+        if (values_file.endswith("'") or values_file.endswith('"')):
+            values_file = values_file[:-1]
+
+    return values_file_provided, values_file
+
+
+def ensure_namespace_cleanup(configuration):
+    api_instance = kube_client.CoreV1Api(kube_client.ApiClient(configuration))
+    timeout = time.time() + 180
+    while True:
+        if time.time() > timeout:
+            telemetry.set_user_fault()
```

Comment:
Just one question, How will cli core recognise user faults?

If cli-core handles all telemetry things, we can remove these telemetry parts, in an another PR, since we have been using this telemetry logs handling since beginning, when we created this extension.

Reply:
The short answer is, if you raise CLIError, cli core will simply record an userfault to Telemtey.

To resolve the posible mis-categorizing problems, we introduced the new error types https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md. The basic idea is: CLI core will use error type to decide whether this is an userfault or failure